cajita/src/Cajita_Partitioner.hpp

```diff
@@ -135,6 +206,94 @@ class DimBlockPartitioner : public BlockPartitioner<NumSpaceDim>
 
         return ranks_per_dim;
     }
+
+    /*!
+      \brief Get the owned number of cells and the global cell offset of the
+      current MPI rank.
+      \param cart_comm The MPI Cartesian communicator for the partitioning.
+      \param global_cells_per_dim The number of global cells in each dimension.
+      \param owned_num_cell (Return) The owned number of cells of the current
+      MPI rank in each dimension.
+      \param global_cell_offset (Return) The global cell offset of the current
+      MPI rank in each dimension
+    */
+    void getOwnedCellInfo(
```

Comment:
I'm sure you've considered this, but can we share more between `ownedCellInfo` and `ownedCellsPerDimension`? Maybe with some private helper functions